app/uk/gov/hmrc/agentmappingfrontend/connectors/MappingConnector.scala

```diff
@@ -48,22 +62,41 @@ class MappingConnector @Inject()(@Named("agent-mapping-baseUrl") baseUrl: URL, h
     new URL(baseUrl, s"/agent-mapping/test-only/mappings/${arn.value}").toString
   }
 
-  private def findUrl(arn: Arn): String = {
-    new URL(baseUrl, s"agent-mapping/mappings/${arn.value}").toString
+  private def findSaUrl(arn: Arn): String = {
+    new URL(baseUrl, s"agent-mapping/mappings/sa/${arn.value}").toString
+  }
+
+  private def findVatUrl(arn: Arn): String = {
+    new URL(baseUrl, s"agent-mapping/mappings/vat/${arn.value}").toString
+  }
+
+  def findSaMappingsFor(arn:Arn)(implicit hc: HeaderCarrier, ec: ExecutionContext): Future[Seq[SaMapping]] = {
+    monitor(s"ConsumedAPI-Mapping-FindSaMappingsForArn-GET") {
+      httpGet.GET[JsValue](findSaUrl(arn)).map {
```

Comment:
Is there a reason you chose to make findSAMappingsFor & findVatMappingsFor separate rather than more generic approach and have it as 1

Reply:
No, no reason, I simply prefer to not complicate unnecessarily, but in the next iteration, we can extract common private method.

Reply:
ideally, since we will be doing this sort of thing for various services, I think we should not repeat our code, and make it more generic and perhaps differentiate them with some parameter(making the fieldName suggestive enough, explaining what it will be used for), as looking at that you can modify this with differentiation in with parameter unless you think it will really over complicate things

Reply:
atm there is no case for this as we only expose those operations via separate test-only endpoints

Reply:
so I guess it would overcomplicate only